src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -0,0 +1,722 @@
+                    parameter->set_friendly_name(input.get_node()->get_friendly_name() + "_" +
+                                                 std::to_string(input.get_index()) + "_parameter");
+                    ov::copy_runtime_info(input.get_node()->shared_from_this(), parameter);
+                    input.replace_source_output(parameter->output(0));
+                    subgraphIds.emplace(parameter.get(), input_subset.first);
+                    subgraphParameterToPrevResult.emplace(parameter.get(), result.get());
+                    _blobNameMap.emplace(parameter->get_default_output().get_any_name(), output.get_any_name());
+                }
+            }
+        }
+    }
+
+    struct Subgraph {
+        ngraph::ResultVector _results;
+        ngraph::ParameterVector _parameters;
+        ngraph::SinkVector _sinks;
+        std::string _affinity;
+    };
+    std::unordered_map<int, Subgraph> subgraphs;
+    // Extracts subgraph parameters, results and affinities
+    for (auto&& subgraphIdPtrValue : subgraphIds) {
+        auto node = subgraphIdPtrValue.first;
+        auto& subgraph = subgraphs[subgraphIdPtrValue.second];
+        if (ov::op::util::is_output(node)) {
+            subgraph._results.emplace_back(std::dynamic_pointer_cast<ngraph::op::v0::Result>(node->shared_from_this()));
+        } else if (ov::op::util::is_parameter(node)) {
+            subgraph._parameters.emplace_back(
+                std::dynamic_pointer_cast<ngraph::op::v0::Parameter>(node->shared_from_this()));
+        } else if (ov::op::util::is_sink(node)) {
+            subgraph._sinks.emplace_back(std::dynamic_pointer_cast<ngraph::op::Sink>(node->shared_from_this()));
+        }
+        auto itAffinity = affinities.find(node);
+        if (itAffinity != affinities.end()) {
+            subgraph._affinity = itAffinity->second;
+        }
+    }
+    results = {};
+
+    // Subgraph topological sort
+    std::vector<Subgraph> allSubgraphs;
+    for (auto&& subgraph : subgraphs) {
+        allSubgraphs.emplace_back(std::move(subgraph.second));
+    }
+
+    std::vector<Subgraph> orderedSubgraphs;
+    NodeSet prevResults;
+    size_t subgraphTopoSortsStep = 0;
+    do {
+        OPENVINO_ASSERT(subgraphTopoSortsStep < subgraphs.size());
+        ++subgraphTopoSortsStep;
+        std::vector<Subgraph> newOrderedSubgraphs;
+        auto IsOrderedSubGraph = [&](const Subgraph& subgraph) {
+            auto& parameters = subgraph._parameters;
+            return std::all_of(parameters.begin(),
+                               parameters.end(),
+                               [&](const ngraph::ParameterVector::value_type& parameter) {
+                                   return InferenceEngine::details::contains(graphInputNodes, parameter.get()) ||
+                                          InferenceEngine::details::contains(prevResults, subgraphParameterToPrevResult[parameter.get()]);
```

Comment:
Please do not use legacy namespaces

Reply:
Will be fixed in https://github.com/openvinotoolkit/openvino/pull/18429